core/trino-main/src/main/java/io/trino/operator/output/UnnestingPositionsAppender.java

```diff
@@ -44,12 +49,14 @@ public void append(IntArrayList positions, Block source)
             return;
         }
         if (source instanceof RunLengthEncodedBlock) {
+            dictionaryBlockBuilder.flushDictionary(delegate);
             delegate.appendRle(((RunLengthEncodedBlock) source).getValue(), positions.size());
         }
```

Comment:
nit1: Add a check in `appendRle` implementations that `Block block` has `<=1` positions.

nit2: In `appendRle` implementations, e.g:
```
    @Override
    public void appendRle(Block block, int rlePositionCount)
    {
        if (rlePositionCount == 0) {
            return;
        }
        int sourcePosition = 0;
```
there is `int sourcePosition = 0;`, just inline it

nit3: javadoc for `io.trino.operator.output.PositionsAppender#appendRle` no longer matches

Reply:
Doesn't seem related to this PR. I can prepare separate change with this in spare time